### PR TITLE
include: add missing zephyr/irq.h include

### PR DESCRIPTION
**Pull request by gmarull**
Change automated searching for files using "IRQ_CONNECT()" API not including <zephyr/irq.h>.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

### CODE REVIEW

**Review by carlocaione — DISMISSED** [review]
LGTM for the two files I own

File: drivers/ethernet/eth_e1000.c [hunk 1]
```diff
@@ -20,6 +20,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #if defined(CONFIG_ETH_E1000_PTP_CLOCK)
 #include <zephyr/drivers/ptp_clock.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/ethernet/eth_e1000.c:23]: need to be outside ifdef

File: drivers/ethernet/eth_mcux.c [hunk 2]
```diff
@@ -52,6 +52,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <zephyr/drivers/gpio.h>
 #if defined(CONFIG_PINCTRL)
 #include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/ethernet/eth_mcux.c:55]: same here

File: drivers/ethernet/eth_smsc911x.c [hunk 3]
```diff
@@ -32,6 +32,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #ifdef CONFIG_SHARED_IRQ
 #include <zephyr/shared_irq.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/ethernet/eth_smsc911x.c:35]: .

File: drivers/ethernet/eth_stm32_hal.c [hunk 4]
```diff
@@ -31,6 +31,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #if defined(CONFIG_PTP_CLOCK_STM32_HAL)
 #include <zephyr/drivers/ptp_clock.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/ethernet/eth_stm32_hal.c:34]: .

File: drivers/flash/flash_stm32_ospi.c [hunk 5]
```diff
@@ -33,6 +33,7 @@ LOG_MODULE_REGISTER(flash_stm32_ospi, CONFIG_FLASH_LOG_LEVEL);
 #include <zephyr/drivers/dma/dma_stm32.h>
 #include <zephyr/drivers/dma.h>
 #include <stm32_ll_dma.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/flash/flash_stm32_ospi.c:36]: need to be outside ifdef

File: drivers/flash/nrf_qspi_nor.c [hunk 6]
```diff
@@ -1388,6 +1388,7 @@ void  z_impl_nrf_qspi_nor_base_clock_div_force(const struct device *dev,
 
 #ifdef CONFIG_USERSPACE
 #include <zephyr/syscall_handler.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/flash/nrf_qspi_nor.c:1391]: .

**Review by cvinayak — DISMISSED** [review]
Approved for `subsys/bluetooth/controller/*` changes.

File: drivers/i2c/i2c_rv32m1_lpi2c.c [hunk 7]
```diff
@@ -16,6 +16,7 @@
 #include <zephyr/logging/log.h>
 #ifdef CONFIG_PINCTRL
 #include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/i2c/i2c_rv32m1_lpi2c.c:19]: .

File: drivers/i2s/i2s_mcux_flexcomm.c [hunk 8]
```diff
@@ -16,6 +16,7 @@
 #include <zephyr/logging/log.h>
 #ifdef CONFIG_PINCTRL
 #include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/i2s/i2s_mcux_flexcomm.c:19]: .

File: drivers/ieee802154/ieee802154_b91.c [hunk 9]
```diff
@@ -23,6 +23,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <zephyr/net/ieee802154_radio.h>
 #if defined(CONFIG_NET_L2_OPENTHREAD)
 #include <zephyr/net/openthread.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/ieee802154/ieee802154_b91.c:26]: .

File: drivers/interrupt_controller/intc_shared_irq.c [hunk 10]
```diff
@@ -16,6 +16,7 @@
 
 #ifdef CONFIG_IOAPIC
 #include <zephyr/drivers/interrupt_controller/ioapic.h>
+#include <zephyr/irq.h>
 #endif
```
Comment (stephanosio) [inline drivers/interrupt_controller/intc_shared_irq.c:20]: .

File: drivers/serial/uart_nrfx_uart.c [hunk 11]
```diff
@@ -17,6 +17,7 @@
 #include <zephyr/drivers/pinctrl.h>
 #else
 #include <hal/nrf_gpio.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/serial/uart_nrfx_uart.c:20]: .

File: drivers/serial/uart_nrfx_uarte.c [hunk 12]
```diff
@@ -35,6 +35,7 @@ LOG_MODULE_REGISTER(uart_nrfx_uarte, CONFIG_UART_LOG_LEVEL);
 #define gppi_channel_enable nrfx_ppi_channel_enable
 #elif defined(CONFIG_HAS_HW_NRF_DPPIC)
 #include <nrfx_dppi.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/serial/uart_nrfx_uarte.c:38]: .

File: drivers/serial/uart_ns16550.c [hunk 13]
```diff
@@ -46,6 +46,7 @@
 #if DT_ANY_INST_ON_BUS_STATUS_OKAY(pcie)
 BUILD_ASSERT(IS_ENABLED(CONFIG_PCIE), "NS16550(s) in DT need CONFIG_PCIE");
 #include <zephyr/drivers/pcie/pcie.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/serial/uart_ns16550.c:49]: .

File: drivers/serial/uart_pl011.c [hunk 14]
```diff
@@ -17,6 +17,7 @@
 
 #ifdef CONFIG_CPU_CORTEX_M
 #include <cmsis_compiler.h>
+#include <zephyr/irq.h>
 #endif
```
Comment (stephanosio) [inline drivers/serial/uart_pl011.c:21]: .

File: drivers/serial/uart_xlnx_ps.c [hunk 15]
```diff
@@ -35,6 +35,7 @@
 
 #ifdef CONFIG_PINCTRL
 #include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/serial/uart_xlnx_ps.c:38]: .

File: drivers/spi/spi_xec_qmspi_ldma.c [hunk 16]
```diff
@@ -26,6 +26,7 @@ LOG_MODULE_REGISTER(spi_xec, CONFIG_SPI_LOG_LEVEL);
 /* #define XEC_QMSPI_DEBUG */
 #ifdef XEC_QMSPI_DEBUG
 #include <zephyr/sys/printk.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline drivers/spi/spi_xec_qmspi_ldma.c:29]: .

File: subsys/testsuite/include/zephyr/interrupt_util.h [hunk 17]
```diff
@@ -110,6 +110,7 @@ static inline void trigger_irq(int irq)
 #define VECTOR_MASK 0xFF
 #else
 #include <zephyr/sys/arch_interface.h>
+#include <zephyr/irq.h>
```
Comment (stephanosio) [inline subsys/testsuite/include/zephyr/interrupt_util.h:113]: .

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: drivers/ethernet/eth_e1000.c (hunk 1 above)
Comment (gmarull) [reply drivers/ethernet/eth_e1000.c]: thanks for spotting all those cases, my script was definitely too simple.

File: drivers/i2c/i2c_rv32m1_lpi2c.c (hunk 7 above)
Comment (stephanosio) [reply drivers/i2c/i2c_rv32m1_lpi2c.c:19]: looks like you missed this one

Comment (gmarull) [reply drivers/i2c/i2c_rv32m1_lpi2c.c]: oops, done

**Review by stephanosio — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Review by firscity — APPROVED** [review]
Xen changes approved.

**Review by jhedberg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by npitre — APPROVED** [review]

**Pull request merged**